tests/Microsoft.Identity.Web.Test/TokenAcquisitionAuthorityTests.cs

```diff
@@ -309,5 +310,86 @@ public void MergeExtraQueryParameters_MergedOptionsNull_Test()
             // Assert
             Assert.Null(mergedDict);
         }
+
+        [Theory]
+        [MemberData(nameof(ManagedIdClientIdCases))]
+        public void ManagedIdCacheKey_Test(string clientId)
+        {
+            // Arrange
+            string defaultKey = ;
+            ManagedIdentityOptions managedIdentityOptions = new()
+            {
+                UserAssignedClientId = clientId
+            };
+
+            // Act
+            string key = GetCacheKeyForManagedIdReflection(managedIdentityOptions);
+
+            // Assert
+            if (string.IsNullOrEmpty(clientId))
+            {
+                Assert.Equal(defaultKey, key);
+            }
+            else
+            {
+                Assert.Equal(clientId, key);
+            }
+        }
+
+
+        [Theory]
+        [MemberData(nameof(ManagedIdClientIdCases))]
+        public void GetOrBuildManagedIdentity_Test(string clientId)
+        {
+            // Arrange
+            ManagedIdentityOptions managedIdentityOptions = new()
+            {
+                UserAssignedClientId = clientId
+            };
+            BuildTheRequiredServices();
+            MergedOptions mergedOptions = new();
+            InitializeTokenAcquisitionObjects();
+            string key = GetCacheKeyForManagedIdReflection(managedIdentityOptions);
+            ConcurrentDictionary<string, IManagedIdentityApplication?> cacheDict = GetManagedIdCacheReflection();
+
+            // Act
+            IManagedIdentityApplication app = 
+                _tokenAcquisition.GetOrBuildManagedIdentityApplication(mergedOptions, managedIdentityOptions).Result;
+
+            // Assert
+            Assert.Equal(app, cacheDict[key]);
```

Comment:
Not a huge fan of this method with using so much reflection. Why do we have to get the private dictionary? Isn't it sufficient to do something like getting calling the GetOrBuild method twice and checking that they are the same object?

Reply:
You're completely right, had too much fun playing with reflection in the previous method and didn't stop to think whether I needed it XD